doc/level0.rst

```diff
@@ -92,26 +92,150 @@ To solve this problem in a better way, bi-directional communication
+++++++
+
+An Autocrypt client MUST NOT create an Autocrypt Setup Message without
+explicit user interaction.  When the user takes this action for a
+specific account, the client:
+
+ * Generates a strong setup code from a CSPRNG.
+ * Optionally, displays the setup code to the user, prompts the user
+   to write it down, and then hides it and asks the user to re-enter
+   it before continuing.  This minor annoyance is a recommended
+   defense against worse annoyance: it ensures that the code was
+   actually written down and the Autocrypt Setup Message is not
+   rendered useless.
+ * Produces an ASCII-armored, minimized OpenPGP transferable secret
+   key out of the key associated with that account.
+ * Symmetrically encrypts the OpenPGP transferable secret key using
+   the secret code as the password.
+ * Composes a new self-addressed e-mail message that contains the
+   payload as a MIME part with the appropriate Content-Type and other
+   headers.
+ * Sends the generated e-mail message to its own account.
+ * Suggests to the user to either back up the message or to import it
+   from another Autocrypt-capable client.
+
+A Level 0 client MUST be able to create an Autocrypt Setup Message, to
+preserve users' ability to recover from disaster, and to choose to use
+a different Autocrypt-capable client in the future.
+
+
+Setup Message Import
+++++++
+
+An Autocrypt-capable client SHOULD support the ability to scan for and
+import an Autocrypt Setup Message when the user has not yet configured
+Autocrypt.  This could happen either when a user of an unconfigured
+Autocrypt client decides to enable Autocrypt, or the client could
+proactively scan the client's mailbox for a message that matches these
+characteristics, and it could alert the client if it discovers one.
+
+If the client finds an Autocrypt Setup Message, it should offer to
+import it to enable Autocrypt.  If the user agrees to do so:
+
+ * The client should prompt the user for their corresponding Setup
+   Code.  If there is an ``Autocrypt-Setup-Message: v1`` OpenPGP
+   header, the client may choose to present the user with a
+   specialized input dialog that better assists the user with input in
+   this particular format.
+ * The client should try decrypting the message with the supplied
+   Setup Code.  If it decrypts:
+ * The client should verify that the User ID on the key matches the
+   User ID on the relevant account.
+ * If it does, the client should import the secret key material and
+   announce to the user that the import was successful.
+
+FIXME: should we delete the message after import?  if the user
+declines to import it, should we delete it?
+
+Why were some of these choices made?
+++++
+
+We chose salted+iterated S2K.  While the use of a memory-hard KDF like
+scrypt or argon2 would be desirable in the future, this is not
+specified in OpenPGP so far, and it is a bigger concern to preserve
+compatibility and avoid friction with presently deployed OpenPGP
+software.
+
```

Comment:
What is missing here is a description of the algorithm chosen and the precise pgp message (like it is done with type=p keys).